stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketObservability.java

```diff
@@ -0,0 +1,387 @@
+    static final String WEBSOCKET_ERROR_TYPE_CONNECTION = "connection";
+    public static final String WEBSOCKET_ERROR_TYPE_CLOSE = "close";
+    public static final String WEBSOCKET_ERROR_TYPE_MESSAGE_SENT = "message_sent";
+    static final String WEBSOCKET_ERROR_TYPE_MESSAGE_RECEIVED = "message_received";
+    static final String WEBSOCKET_ERROR_TYPE_UNEXPECTED = "unexpected";
+
+    private static final String WEBSOCKET_UNKNOWN = "unknown";
+
+    /**
+     * This method observes all incoming WebSocket connection requests.
+     *
+     * @param connectionInfo information regarding connection.
+     * @param result whether the connection was successful or not.
+     */
+    static void observeRequest(WebSocketOpenConnectionInfo connectionInfo, String result) {
+        if (ObserveUtils.isObservabilityEnabled()) {
+            ObserverContext observerContext = new ObserverContext();
+            observerContext.setConnectorName(SERVER_CONNECTOR_WEBSOCKET);
+            try {
+                observerContext.addTag(TAG_CONNECTION_ID, connectionInfo.getWebSocketConnection().getChannelId());
+            } catch (IllegalAccessException e) {
+                observerContext.addTag(TAG_CONNECTION_ID, WEBSOCKET_UNKNOWN);
+            }
+            setObserveService(observerContext, connectionInfo);
+            observerContext.addTag(TAG_KEY_RESULT, result);
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
+
+            MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
+            //Increment requests metric
+            metricRegistry.counter(new MetricId(SERVER_CONNECTOR_WEBSOCKET + "_" + METRIC_REQUESTS,
+                                                METRIC_REQUESTS_DESC, allTags)).increment();
+
+            //Log request
+            logger.info("WS connection request received");
+        }
+    }
+
+    /**
+     * This method observes successful WebSocket connections.
+     *
+     * @param connectionInfo information regarding connection.
+     */
+    static void observeConnection(WebSocketOpenConnectionInfo connectionInfo) {
+        if (ObserveUtils.isObservabilityEnabled()) {
+            ObserverContext observerContext = new ObserverContext();
+            observerContext.setConnectorName(SERVER_CONNECTOR_WEBSOCKET);
+            try {
+                observerContext.addTag(TAG_CONNECTION_ID,
+                                       connectionInfo.getWebSocketConnection().getChannelId());
+            } catch (IllegalAccessException e) {
+                observerContext.addTag(TAG_CONNECTION_ID, WEBSOCKET_UNKNOWN);
+            }
+            setObserveService(observerContext, connectionInfo);
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
```

Comment:
I think most of this code is same as the previous function. Shall we remove duplicates by creating a new function?